vimrc

```diff
@@ -106,8 +106,18 @@ inoremap <S-Tab> <c-n>
 " Exclude Javascript files in :Rtags via rails.vim due to warnings when parsing
 let g:Tlist_Ctags_Cmd="ctags --exclude='*.js'"
 
+
 " Index ctags from any project, including those outside Rails
-map <Leader>ct :!ctags -R .<CR>
+function! ReindexCtags()
+  let l:ctags_hook_location = '$(git rev-parse --show-toplevel)/.git/hooks/ctags'
+
+  if exists(l:ctags_hook_location)
+    exec '!'. l:ctags_hook_location
+  else
+    exec "!ctags -R ."
+  endif
+endfunction
+nmap <Leader>ct :call ReindexCtags()<CR>
```

Comment:
I split my vimrc up by placing plugin configurations into `.vim/plugin/settings`, and extracting plugins like this one into `.vim/plugin`. That way, there is nothing that I need to source explicitly.